common/record/record.js

```diff
@@ -212,16 +229,84 @@ const deleteNode = node =>
       newValidation => Validator.assocValidation(newValidation)(recordUpdated)
     )(recordUpdated)
 
+    const recordCacheUpdated = removeNodeFromIndex(node)(recordValidationUpdated)
+
     // 3. remove entity children recursively
-    const children = getNodeChildren(node)(recordValidationUpdated)
+    const children = getNodeChildren(node)(recordCacheUpdated)
 
     return R.reduce(
       (recordCurrent, child) => deleteNode(child)(recordCurrent),
-      recordValidationUpdated,
+      recordCacheUpdated,
       children
     )
   }
 
+// ======= CACHE
+
```

Comment:
Record.js is becoming huge, would it make sense to move the cache part into a separate file, e.g. RecordCache or .. (that is accessible only from within record.js)?

Reply:
yes we can